architecture/decisions/0012-we-chose-a-retry-strategy.md

```diff
@@ -0,0 +1,47 @@
+# 11. We have chosen a retry strategy in case of processing errors
+
+Date: 2018-02-06
+
+## State
+
+Accepted
+
+## Context
+
+To perform asynchronous operations we use [Azure Storage Queues](https://azure.microsoft.com/en-us/services/storage/queues/)):
+enqueued messages are processed by the methods exposed by Azure Functions using [triggers](https://docs.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings.
+
+In case some error occurs it is necessary to re-schedule the message in the queue
+to be processed at a later time.
+
+Our goal is to re-schedule the processing at increasing intervals, using an exponential back off strategy for retries.
+
+The default behavior is specified in the [triggers documentation](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue#trigger---poison-messages):
+
+> When a queue trigger function fails, the Azure functions retry the function up to five times for a given queued message, including the first attempt. 
+> If all five attempts fail, the runtime> functions adds a message to a queue called <originalqueuename> -poison.
+
+Currently the Function SDK does not provide an *output binding* to delay the processing of an enqueued message.
+It is therefore necessary to implement a custom retry policy to schedule messages over a longer period of time.
+
+## Decision
+
+The following possibilities, to delay message processing, have been considered:
```

Comment:
I would move the considered possibilities in the "context" section as the "decision" section is usually focused on the actual action taken